src/components/Providers/SystemIOProviderWeb.tsx

```diff
@@ -0,0 +1,52 @@
+import { useEffect, useCallback } from 'react'
+import { commandBarActor } from '@src/machines/commandBarMachine'
+import { importFileFromURL } from '@src/lib/commandBarConfigs/projectsCommandConfig'
+import { useClearURLParams } from '@src/machines/systemIO/hooks'
+import { useSearchParams } from 'react-router-dom'
+import { CREATE_FILE_URL_PARAM } from '@src/lib/constants'
+
+export function SystemIOMachineLogicListenerWeb() {
+  const clearURLParams = useClearURLParams()
+  const [searchParams, setSearchParams] = useSearchParams()
+  const clearImportSearchParams = useCallback(() => {
+    // Clear the search parameters related to the "Import file from URL" command
+    // or we'll never be able cancel or submit it.
+    searchParams.delete(CREATE_FILE_URL_PARAM)
+    searchParams.delete('code')
+    searchParams.delete('name')
+    searchParams.delete('units')
+    setSearchParams(searchParams)
+  }, [searchParams, setSearchParams])
+
+  const useAddProjectCommandsToCommandBar = () => {
+    const commands = [importFileFromURL]
+    useEffect(() => {
+      commandBarActor.send({
+        type: 'Add commands',
+        data: {
+          commands: commands,
+        },
+      })
+      return () => {
+        commandBarActor.send({
+          type: 'Remove commands',
+          data: {
+            commands: commands,
+          },
+        })
+      }
+    }, [])
+  }
```

Comment:
I think you can get rid of these hooks by adding the commands directly to the palette. I've made a branch off yours here that you can check if you like: https://github.com/KittyCAD/modeling-app/compare/nadro/adhoc/system-io-machine...franknoirot/nadro-6352?expand=1

Reply:
Perfect that is even better, thanks for telling me about that pattern. I've merged your branch and pushed the code to this branch. 

```
commit 794568f8583afb208b2469a4ba480a2c216a2511 (HEAD -> nadro/adhoc/system-io-machine, origin/nadro/adhoc/system-io-machine)
Merge: 9d14be7e1 7255dea87
Author: Kevin Nadro <kevin@zoo.dev>
Date:   Wed Apr 23 08:32:38 2025 -0500

    Merge branch 'franknoirot/nadro-6352' into nadro/adhoc/system-io-machine
```